services/wireguard/connection/connection.go

```diff
@@ -105,6 +111,23 @@ func (c *Connection) Stop() {
 	}
 
 	c.stateChannel <- connection.NotConnected
-	c.connection.Done()
+	c.cancel()
```

Comment:
might want to close the statistics channel here, to inform the consumer that we're not gonna send anymore